File: JH/DAY14/1260.cpp
#include <bits/stdc++.h>

using namespace std;


vector<int>graph[10002];
int visited[10002];

void dfs(int x){

    visited[x] = 1 ;
    cout << x << " ";

    for (int i = 0; i <graph[x].size() ; ++i) {
        int nx = graph[x][i];
        if(visited[nx]==0){
            dfs(nx);
        }
    }
}

void bfs(int x){
    queue<int>q;
    q.push(x);
    visited[x]=1;
    while(!q.empty()){
        int now = q.front();
        q.pop();
        cout << now << " ";
        for (int i = 0; i < graph[now].size() ; ++i) {
            int nx = graph[now][i];
            if(visited[nx]==0){
                visited[nx]=1;
                q.push( nx);
            }
        }
    }
}




int main(){

    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    int n;
    int m;
    int v;

    cin >> n >> m >> v;

    for (int i = 0; i <m ; ++i) {
        int a, b;
        cin >> a>> b;
        graph[a].push_back(b);
        graph[b].push_back(a);
    }

    for (int i = 1; i <= n;i++){
        sort(graph[i].begin(), graph[i].end()); // 낮은 숫자부터 탐색.
    }
    dfs(v);
    cout << "\n";
    memset(visited,0,10002);
    bfs(v);














}

File: KSY/DAY12/1260.py
import queue

N, M, V = map(int, input().split())
edges = {}

stack = [V] #DFS
queue = queue.Queue() #BFS
queue.put(V)

#DFS
for _ in range(M):
    v1, v2 = map(int, input().split())
    edges[v1] = edges.get(v1, []) + [v2]
    edges[v2] = edges.get(v2, []) + [v1]
    # print(edges[v1])

answer = ""
visited = set()
while stack:
    s = stack.pop()
    if s not in visited:
        answer += str(s) + " "
        visited.add(s)

        if s in edges:
            edges[s].sort(reverse=True) #큰 수부터 stack에 넣어 작은 수 부터 꺼내기 위함
            for _s in edges[s]:
                if _s:
                    stack.append(_s)

print(answer)

answer = ""
visited = set()
while not queue.empty(): #queue는 list처럼 내용이 없으면 0을 뱉지 않음.
    q = queue.get()
    if q not in visited:
        answer += str(q) + " "
        visited.add(q)

        if q in edges:
            edges[q].sort()
            for _q in edges[q]:
                queue.put(_q)

print(answer)